core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -861,6 +861,16 @@ private[spark] class SparkSubmit extends Logging {
    * running cluster deploy mode or python applications.
    */
   private def runMain(args: SparkSubmitArguments, uninitLog: Boolean): Unit = {
+    // set the handler as early as possible
+    // we need the driver check because spark-submit is used by mesos to launch
+    // executors (via the SparkLauncher)
+    val isExecutor = args.childArgs.contains("--executor-id")
```

Comment:
no the best but at this point this is what is available (the alternative is to add it at the SparkContext side)